aea/protocols/dialogue/base.py

```diff
@@ -188,7 +188,31 @@ def from_str(cls, obj: str) -> "DialogueLabel":
         return dialogue_label
 
 
-class Dialogue(ABC):
+class _DialogueMeta(type):
```

Comment:
Open questions:
- Memory footprint of extensions (additional attributes) using slots vs extensions using body (lazy instantiated). There is an obvious LOC saving with the latter approach over the former. (high priority)
- Describe alternative approach to remove additional attributes from dialogue. (low priority)